elastic/kibana
# [Alerting] Remove defaultRuleTaskTimeout and set ruleType specific timeout from kibana.yml

fixes: #124807

doc preview: https://kibana_128294.docs-preview.app.elstc.co/guide/en/kibana/master/alert-action-settings-kb.html#alert-settings

## Summary

This PR intends to remove `defaultRuleTaskTimeout` config and use `config.execution.timeout` key instead of it.
The new key `config.execution.timeout` can be modified by the users globally or per rule type.

## To Verify
- Add a sleep function into the task runner (just before `this.ruleType.executor`) code such as: `await new Promise(r => setTimeout(r, 360000));`
   That would delay your rule execution for 6 minutes.
- Add the below config to your kibana.yml and run kibana
- Create two rules (e.g. Index Threshold and a Security Rule) and wait your rules to be executed.

 Index Threshold should timeout in 10s and the Security Rule in 5m.
 You can see the error message in the rule detail page with the applied timeout config.
 
```
xpack.alerting.rules.execution:
    timeout: 30s
    ruleTypeOverrides:
      - id: '.index-threshold'
         timeout: 10s
```

To try global timeout config remove `ruleTypeOverrides` from your kibana.yml and run the kibana again...
You should see 30s in the error message on the rule details page.
 
```
xpack.alerting.rules.execution:
    timeout: 30s
```

### Checklist

Delete any items that are not applicable to this PR.
